internal/pkg/acl/acl_windows_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build windows
+
+package acl
+
+import (
+	"os"
+	"runtime"
+	"testing"
+
+	"golang.org/x/sys/windows"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestChmod(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip("skipping test on non-windows platform")
+	}
+
+	// Create a temporary file
+	tmpFile, err := os.CreateTemp("", "testfile")
```

Comment:
```suggestion
	tmpFile, err := os.CreateTemp(t.TempDir(), "testfile")
```

The TempDir is automatically deleted, so you don't have to worry about the cleanup and more importantly it has [removeAll](https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/testing/testing.go;drc=3aa7c5ef01e147fb482f4b3e79c6f875a4b1b9fb;l=1290) that handles Windows giving a temporary "Access Denied" error message.

Reply:
Hit the wrong button, should be a comment so optional.  Switching to `t.TempDir` is shorter, uses `testing` package in a more idiomatic way and has more flaky test protections.  Up to you if you want to switch.

Reply:
ok if this comes up as I flaky I can provide guarantees that I will open such a fix on the next instant 🙂